### PR TITLE
Add a minimal example for solving a system

### PR DESCRIPTION
**Pull request by gflegar**
This PR adds the minimal possible example that solves a system on the GPU.
It is the same example that is going to be included into the interface paper, so it's probably a good idea to also have it here.

### CODE REVIEW

**Comment by hartwiganzt** [comment]
LGTM. But I am not sure about the name...

File: examples/minimalistic_solver/minimalistic_solver.cpp [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    // Create the solver
+    auto solver =
+        gko::solver::Cg<>::Factory::create()
+            .with_preconditioner(
+                gko::preconditioner::BlockJacobiFactory<>::create(exec, 32))
+            .with_criterion(
+                gko::stop::Combined::Factory::create()
+                    .with_criteria(
+                        gko::stop::Iteration::Factory::create()
+                            .with_max_iters(20u)
```
Comment (hartwiganzt) [inline examples/minimalistic_solver/minimalistic_solver.cpp]: I did not notice before we write "(20u)" for 20 iterations. what is the u for?

Comment (tcojean) [reply examples/minimalistic_solver/minimalistic_solver.cpp]: The `u` is for unsigned. Due to `max_iters` being `size_type` which is typically some unsigned number under the hood, you cannot simply type `20`, you have to make sure the number is unsigned, either with static cast or adding the `u` suffix as is done here.

Comment (hartwiganzt) [reply examples/minimalistic_solver/minimalistic_solver.cpp]: thanks! is that documented somewhere?

Comment (tcojean) [reply examples/minimalistic_solver/minimalistic_solver.cpp]: Well the compiler gives you an error and all examples and test use this. But I don't think I added a comment specifically about that anywhere. I can add it.

**Comment by hartwiganzt** [comment]
how about minimal_solver_gpu ? Because it is hard-coded GPU, right?

**Comment by gflegar** [comment]
Renamed to `minimal_solver_cuda`.

**Comment by gflegar** [comment]
Merging something from private repo first - don't merge this yet to avoid conflicts.

**Pull request merged**